config.m4

```diff
@@ -46,54 +55,53 @@ if test "$PHP_MONGODB" != "no"; then
+    AX_CHECK_COMPILE_FLAG(-Wno-sign-compare,                PHP_MONGODB_DEV_CFLAGS="$PHP_MONGODB_DEV_CFLAGS -Wno-sign-compare"              ,, -Werror)
+    AX_CHECK_COMPILE_FLAG(-fstack-protector,                PHP_MONGODB_DEV_CFLAGS="$PHP_MONGODB_DEV_CFLAGS -fstack-protector"              ,, -Werror)
+    AX_CHECK_COMPILE_FLAG(-fno-exceptions,                  PHP_MONGODB_DEV_CFLAGS="$PHP_MONGODB_DEV_CFLAGS -fno-exceptions"                ,, -Werror)
+    AX_CHECK_COMPILE_FLAG(-Wformat-security,                PHP_MONGODB_DEV_CFLAGS="$PHP_MONGODB_DEV_CFLAGS -Wformat-security"              ,, -Werror)
+    AX_CHECK_COMPILE_FLAG(-Wformat-nonliteral,              PHP_MONGODB_DEV_CFLAGS="$PHP_MONGODB_DEV_CFLAGS -Wformat-nonliteral"            ,, -Werror)
+    AX_CHECK_COMPILE_FLAG(-Winit-self,                      PHP_MONGODB_DEV_CFLAGS="$PHP_MONGODB_DEV_CFLAGS -Winit-self"                    ,, -Werror)
+    AX_CHECK_COMPILE_FLAG(-Wwrite-strings,                  PHP_MONGODB_DEV_CFLAGS="$PHP_MONGODB_DEV_CFLAGS -Wwrite-strings"                ,, -Werror)
+    AX_CHECK_COMPILE_FLAG(-Wenum-compare,                   PHP_MONGODB_DEV_CFLAGS="$PHP_MONGODB_DEV_CFLAGS -Wenum-compare"                 ,, -Werror)
+    AX_CHECK_COMPILE_FLAG(-Wempty-body,                     PHP_MONGODB_DEV_CFLAGS="$PHP_MONGODB_DEV_CFLAGS -Wempty-body"                   ,, -Werror)
+    AX_CHECK_COMPILE_FLAG(-Wparentheses,                    PHP_MONGODB_DEV_CFLAGS="$PHP_MONGODB_DEV_CFLAGS -Wparentheses"                  ,, -Werror)
+    AX_CHECK_COMPILE_FLAG(-Wmaybe-uninitialized,            PHP_MONGODB_DEV_CFLAGS="$PHP_MONGODB_DEV_CFLAGS -Wmaybe-uninitialized"          ,, -Werror)
+    AX_CHECK_COMPILE_FLAG(-Wimplicit-fallthrough,           PHP_MONGODB_DEV_CFLAGS="$PHP_MONGODB_DEV_CFLAGS -Wimplicit-fallthrough"         ,, -Werror)
+    AX_CHECK_COMPILE_FLAG(-Werror,                          PHP_MONGODB_DEV_CFLAGS="$PHP_MONGODB_DEV_CFLAGS -Werror"                        ,, -Werror)
+    AX_CHECK_COMPILE_FLAG(-Wextra,                          PHP_MONGODB_DEV_CFLAGS="$PHP_MONGODB_DEV_CFLAGS -Wextra"                        ,, -Werror)
+    AX_CHECK_COMPILE_FLAG(-Wno-unused-parameter,            PHP_MONGODB_DEV_CFLAGS="$PHP_MONGODB_DEV_CFLAGS -Wno-unused-parameter"          ,, -Werror)
+    AX_CHECK_COMPILE_FLAG(-Wno-unused-but-set-variable,     PHP_MONGODB_DEV_CFLAGS="$PHP_MONGODB_DEV_CFLAGS -Wno-unused-but-set-variable"   ,, -Werror)
+    AX_CHECK_COMPILE_FLAG(-Wno-missing-field-initializers,  PHP_MONGODB_DEV_CFLAGS="$PHP_MONGODB_DEV_CFLAGS -Wno-missing-field-initializers",, -Werror)
 
     if test "$PHP_MONGODB_PHP_VERSION_ID" -ge "80108"; then
-      AX_CHECK_COMPILE_FLAG(-Wstrict-prototypes, _MAINTAINER_CFLAGS="$_MAINTAINER_CFLAGS -Wstrict-prototypes",, -Werror)
+      AX_CHECK_COMPILE_FLAG(-Wstrict-prototypes, PHP_MONGODB_DEV_CFLAGS="$PHP_MONGODB_DEV_CFLAGS -Wstrict-prototypes",, -Werror)
     fi
-
-    MAINTAINER_CFLAGS="$_MAINTAINER_CFLAGS"
```

Comment:
The reasons for these names are lost to time, but since I'm now initializing `$PHP_MONGODB_DEV_CFLAGS` and appending to it directly we can do without this.